pkg/tso/keyspace_group_manager.go

```diff
@@ -410,9 +416,26 @@ func (kgm *KeyspaceGroupManager) updateKeyspaceGroup(group *endpoint.KeyspaceGro
 		return
 	}
 
+	oldAM, oldGroup := kgm.getKeyspaceGroupMeta(group.ID)
+	// If this host owns a replica of the keyspace group which is the merge target,
+	// it should run the merging checker when the merge state first time changes.
+	if !oldGroup.IsMergeTarget() && group.IsMergeTarget() {
+		ctx, cancel := context.WithCancel(kgm.ctx)
+		kgm.mergeCheckerCancelMap[group.ID] = cancel
```

Comment:
Will it be called in multiple places in the same time?

Reply:
Yes, it will be called in each assigned replica and keep running until the merging is finished by the primary node. This is to prevent leader changing to make sure the merge could eventually be done by every node.

Reply:
So does the map have concurrent read/write operations?

Reply:
Since the `updateKeyspaceGroup` will only be called when initializing and the etcd watching event coming, it will always be called sequentially. So it won't have concurrent read/write I think.

Reply:
The watch will start first, does it have a time window that https://github.com/tikv/pd/pull/6625/files#diff-5b6926e1bf5eeb6e97e1b27cb6eb8156d3d06ba9e72fb027f613c407f3674c68R364 happens at the same time?